corehq/apps/case_importer_v1/util.py

```diff
@@ -104,131 +104,40 @@ def open_workbook(filename=None, **kwargs):
-                return int(cell.value)
-            else:
-                return cell.value
-        elif cell.ctype == xlrd.XL_CELL_DATE:
-            return date(*xlrd.xldate_as_tuple(cell.value, self.workbook.datemode)[:3])
-        else:
-            return cell.value
-
-    def _get_first_sheet(self):
-        return self.workbook.sheet_by_index(0)
-
-    def get_header_columns(self):
-        sheet = self._get_first_sheet()
-
-        if sheet.ncols > 0:
-            columns = []
-
-            # get columns
-            if self.column_headers:
-                columns = self._row_values(sheet, 0)
-            else:
-                for colnum in range(sheet.ncols):
-                    columns.append("Column %i" % (colnum,))
-
-            return columns
-        else:
-            return []
-
-    def _get_column_values(self, column_index):
-        sheet = self._get_first_sheet()
-
-        if self.column_headers:
-            return self._col_values(sheet, column_index)[1:]
-        else:
-            return self._col_values(sheet, column_index)
-
-    def get_unique_column_values(self, column_index):
-        return list(set(self._get_column_values(column_index)))
-
-    def _get_num_rows(self):
-        sheet = self._get_first_sheet()
-        return sheet.nrows
-
-    def _get_row(self, index):
-        sheet = self._get_first_sheet()
-        return self._row_values(sheet, index)
-
-    @property
-    def max_row(self):
-        return self._get_num_rows()
-
-    def iter_rows(self):
-        row_count = self.max_row
-        for i in range(row_count):
-            yield self._get_row(i)
-
-
 class WorksheetWrapper(object):
 
-    def __init__(self, worksheet):
-        """
-        raises ImporterError or one of its subtypes
-        if there's a problem with the excel file
-
-        """
-        self.worksheet = worksheet
+    def __init__(self, worksheet, column_headers):
+        self._worksheet = worksheet
+        self._column_headers = column_headers
 
     @classmethod
-    def from_workbook(cls, workbook):
+    def from_workbook(cls, workbook, column_headers):
         if not isinstance(workbook, Workbook):
             raise AssertionError(
                 "WorksheetWrapper.from_workbook called without Workbook object")
         elif not workbook.worksheets:
             raise AssertionError(
                 "WorksheetWrapper.from_workbook called with Workbook with no sheets")
         else:
-            return cls(workbook.worksheets[0])
+            return cls(workbook.worksheets[0], column_headers)
 
     def get_header_columns(self):
-        if self.worksheet.max_row > 0:
-            return self.iter_rows().next()
+        if self.max_row > 0:
+            if self._column_headers:
+                return self.iter_rows().next()
+            else:
+                columns = []
+                for i in range(self.max_row):
+                    columns.append("Column {!d}".format(i))
```

Comment:
I don't think `!d` is a [valid conversion](https://docs.python.org/2.7/library/string.html#formatstrings) . Since `i` is an integer anyway, I think you can just do `{}`
Additional nit, you could use an array comprehension for this.